subtest/scenario.py

```diff
@@ -219,18 +201,15 @@ def __init__(self, params, experiment_path=Path(r"./experiments"), scenario_id=1
         self.scenario_id = scenario_id
         self.n_repeat = n_repeat
 
-        if "is_quick_demo" in params:
-            self.is_quick_demo = params["is_quick_demo"]
-            if self.is_quick_demo and self.dataset_proportion < 1:
-                raise Exception("Don't start a quick_demo without the full dataset")
-        else:
-            self.is_quick_demo = False  # default
+        self.is_quick_demo = is_quick_demo
+        if self.is_quick_demo and self.dataset_proportion < 1:
+            raise Exception("Don't start a quick_demo without the full dataset")
 
         # The quick demo parameters overwrites previously defined parameters to make the scenario faster to compute
-        if "is_quick_demo" in params and params["is_quick_demo"]:
+        if self.is_quick_demo:
             # Use less data and/or less epochs to speed up the computations
             logger.info("Quick demo: limit number of data and number of epochs.")
-            if (len(self.dataset.x_train) > 1000):
+            if len(self.dataset.x_train) > 1000:
                 index_train = np.random.choice(self.dataset.x_train.shape[0], 1000, replace=False)
                 index_val = np.random.choice(self.dataset.x_val.shape[0], 500, replace=False)
                 index_test = np.random.choice(self.dataset.x_test.shape[0], 500, replace=False)
```

Comment:
These hardcoded figures are arbitrary, so I am thinking we should probably put them in `constants`